site2/docs/develop-plugin.md

```diff
@@ -0,0 +1,123 @@
+
+1. Add a dependency for Pulsar broker in the `pom.xml` file to display. Otherwise, you can not find the [`EntryFilter` interface](https://github.com/apache/pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/plugin/EntryFilter.java).
+
+    ```xml
+    <dependency>
+    <groupId>org.apache.pulsar</groupId>
+    <artifactId>pulsar-broker</artifactId>
+    <version>${pulsar.version}</version>
+    <scope>provided</scope>
+    </dependency>
+    ```
+
+2. Implement the  [`FilterResult filterEntry(Entry entry, FilterContext context);` method](https://github.com/apache/pulsar/blob/2adb6661d5b82c5705ee00ce3ebc9941c99635d5/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/plugin/EntryFilter.java#L34).
+
+   - If the method returns `ACCEPT` or NULL, this message is sent to consumers.
+
+   - If the method returns `REJECT`, this message is filtered out and it does not consume message permits. 
+
+   - If there are multiple entry filters, this message passes through all filters in the pipeline in a round-robin manner. If any entry filter returns `REJECT`, this message is discarded.
+
+    You can get entry metadata, subscriptions, and other information through `FilterContext`.
+
+3. Describe a NAR file.
+
+    Create an `entry_filter.yaml` file in the `resources/META-INF/services` directory to describe NAR.
+
+    ```conf
+    # Entry filter name, which should be configured in the broker.conf file later
+    name: entryFilter
+    # Entry filter description
+    description: entry filter
+    # Implementation class name of entry filter 
+    entryFilterClass: com.xxxx.xxxx.xxxx.DefaultEntryFilterImpl
+    ```
+
+#### Step 3: package implementation class of entry filter into NAR
+
+1. Add the compiled plugin of the NAR file to your `pom.xml` file.
+
+    ```xml
+    <build>
+            <finalName>${project.artifactId}</finalName>
+            <plugins>
+                <plugin>
+                    <groupId>org.apache.nifi</groupId>
+                    <artifactId>nifi-nar-maven-plugin</artifactId>
+                    <version>1.2.0</version>
+                    <extensions>true</extensions>
+                    <configuration>
+                        <finalName>${project.artifactId}-${project.version}</finalName>
+                    </configuration>
+                    <executions>
+                        <execution>
+                            <id>default-nar</id>
+                            <phase>package</phase>
+                            <goals>
+                                <goal>nar</goal>
+                            </goals>
+                        </execution>
+                    </executions>
+                </plugin>
+            </plugins>
+        </build>
+    ```
+
+2. Generate a NAR file in the `target` directory.
+
+    ```script
+    mvn clean install
+    ```
+
+#### Step 4: configure and restart broker
+
+1. Configure the following parameters in the `broker.conf` file.
```

Comment:
if it is a local pulsar cluster, should we need to configure the `standalone.conf` file?

Reply:
yes, added